movies.py

```diff
@@ -15,3 +15,10 @@ class Movie(BaseModel):
 
 async def get_all_movies() -> list[Movie]:
     return movies
+
+
+async def get_movie_by_name(name: str) -> Movie | None:
+    try:
+        return next(filter(lambda movie: movie.name == name, await get_all_movies()))
```

Comment:
I try to avoid using `map`/`filter` with ad-hoc lambdas, a generator comprehension is usually more readable.
Also, `next` accepts a second argument in case the iterator is empty, so you can get rid of the `try` and rewrite the entire thing to:
```python
return next((movie for movie in await get_all_movies() if movie.name == name), None)
```

Reply:
The default of the default of `next` is None and in this case it throws the exception 💩

Reply:
That's wrong, just tested it.
(it is possible to differentiate between `None` and `argument not passed` using `*args`, a non-`None` sentinel as a default value or C-level shenanigans, which is what `next` does)

Reply:
<img width="670" alt="Screen Shot 2022-10-04 at 14 45 38" src="https://user-images.githubusercontent.com/16135381/193811764-74d64b84-41f0-4dc2-b5db-b47df0209c53.png">

Pycharm screwed me... didn't read the fine print :)